ChildProject/pipelines/metrics.py

```diff
@@ -510,9 +510,14 @@ def __init__(
     def _process_unit(self, unit: str):
         annotations, segments = self.retrieve_segments([self.set], unit)
 
+        # retrieve timestamps for each vocalization, ignoring the day of occurence
         segments = self.am.get_segments_timestamps(segments, ignore_date=True)
+
+        # dropping segments for which no time information is available
         segments.dropna(subset=["onset_time"], inplace=True)
 
+        # update the timestamps so that all vocalizations appear
+        # to happen on the same day
```

Comment:
does that mean --- put them all in a single time line, consecutive, so that the user gets numbers every eg 5 mins to the end of the rec?

Reply:
this really is just a technicality to allow pandas.Grouper to group together all vocs that happen between e.g. 08:00 and 09:00 (in case of 1-hour periods) regardless of the day they happened. This is useful for recordings lastin > 24h (which do not exist yet), but much more importantly, this is required for recordings that end after midnight.